### PR TITLE
[TwigComponent][Doc] Twig code style & "boolean props" part

### PR DESCRIPTION
**Pull request by smnandre**
Fix code style in Twig examples (to follow twig & symfony code conventions)

Also rewrite/improved the paragraph about "boolean props"
* explain how it works and why
* increase visibility of the solutions

(it's been some time I wanted to rewrite this part, as we often have questions)

### CODE REVIEW

**Review by kbond — APPROVED** [review]
The emoji's are a nice touch!

File: src/TwigComponent/doc/index.rst [hunk 1]
```diff
@@ -802,7 +807,7 @@ Remember though that the ``outerScope`` reference only starts once you're INSIDE
     {# templates/FancyProfileCard.html.twig #}
     {% component Card %}
         {% block header %}
-            {% component Alert with { message: outerScope.this.someProp } %} {# not yet INSIDE the Alert template #}
+            {% component Alert with {message: outerScope.this.someProp } %} {# not yet INSIDE the Alert template #}
```
Comment (WebMamba) [inline src/TwigComponent/doc/index.rst]:
```suggestion
            {% component Alert with {message: outerScope.this.someProp} %} {# not yet INSIDE the Alert template #}
```

File: src/TwigComponent/doc/index.rst [hunk 2]
```diff
@@ -852,7 +859,7 @@ We already have a generic ``Alert`` component, so let's re-use it:
 .. code-block:: html+twig
 
     {# templates/components/Alert.html.twig #}
-    <div {{ attributes.defaults({ class: 'alert alert-'~ type }) }}">
+    <div {{ attributes.defaults({class: 'alert alert-'~ type}) }}">
```
Comment (WebMamba) [inline src/TwigComponent/doc/index.rst]: We have different conventions with the ~ , What do you think about not  adding space around it:

```suggestion
    <div {{ attributes.defaults({class: 'alert alert-'~type}) }}">
```

File: src/TwigComponent/doc/index.rst [hunk 3]
```diff
@@ -929,7 +936,7 @@ To add a custom `Stimulus controller`_ to your root component element:
 
 .. code-block:: html+twig
 
-    <div {{ attributes.defaults(stimulus_controller('my-controller', { someValue: 'foo' })) }}>
+    <div  {{ attributes.defaults(stimulus_controller('my-controller', {someValue: 'foo'})) }}>
```
Comment (WebMamba) [inline src/TwigComponent/doc/index.rst]:
```suggestion
    <div {{ attributes.defaults(stimulus_controller('my-controller', {someValue: 'foo'})) }}>
```

File: src/TwigComponent/doc/index.rst (hunk 2 above)
Comment (smnandre) [reply src/TwigComponent/doc/index.rst]: Thanks!

**Comment by smnandre** [comment]
Thanks @WebMamba for the corrections :)

**Review by Kocal — APPROVED** [review]

**Pull request merged**